src/client/auth/mod.rs

```diff
@@ -140,6 +139,23 @@ impl AuthMechanism {
 
                 Ok(())
             }
+            AuthMechanism::Plain => {
+                if credential.username.is_none() {
```

Comment:
The spec indicates that the username needs to be non-zero length -- is that something we should be validating here as well?

Reply:
Oh, I missed that; we should definitely be validating it. Good catch!